packages/round-manager/src/features/api/round.ts

```diff
@@ -300,57 +308,80 @@ export async function listRounds(
-      _roundFactoryContract.address!,
-      _roundFactoryContract.abi,
-      signerOrProvider,
+    const roundFactoryAddress = roundFactoryContract(chainId);
+    const roundFactory = RoundFactory__factory.connect(
+      roundFactoryAddress,
+      signerOrProvider
     );
 
-    if (!round.applicationsEndTime) {
-      round.applicationsEndTime = round.roundStartTime;
+    let initRoundTimes: string[] = [];
+    const parseDate = (date: Date) =>
+      (new Date(date).getTime() / 1000).toString();
+    if (isQF) {
+      if (!round.applicationsEndTime) {
+        round.applicationsEndTime = round.roundStartTime;
+      }
+      initRoundTimes = [
+        parseDate(round.applicationsStartTime),
+        parseDate(round.applicationsEndTime),
+        parseDate(round.roundStartTime),
+        parseDate(round.roundEndTime as Date),
+      ];
+    } else {
+      // note: DirectRounds does not set application dates.
+      // in those cases, we set:
+      // application start time with the round start time
+      // application end time with MaxUint256.
+      // if the round has not end time, we set with MaxUint256.
+
+      initRoundTimes = [
+        parseDate(round.applicationsStartTime ?? round.roundStartTime),
+        round.applicationsEndTime
+          ? parseDate(round.applicationsEndTime)
+          : round.roundEndTime
+          ? parseDate(round.roundEndTime)
+          : ethers.constants.MaxUint256.toString(),
+        parseDate(round.roundStartTime),
+        round.roundEndTime
+          ? parseDate(round.roundEndTime)
+          : ethers.constants.MaxUint256.toString(),
+      ];
     }
 
     round.operatorWallets = round.operatorWallets?.filter((e) => e !== "");
-
     const initAddresses = [round.votingStrategy, round.payoutStrategy.id];
-
-    const initRoundTimes = [
-      new Date(round.applicationsStartTime).getTime() / 1000,
-      new Date(round.applicationsEndTime).getTime() / 1000,
-      new Date(round.roundStartTime).getTime() / 1000,
-      new Date(round.roundEndTime).getTime() / 1000,
-    ];
-
     const initMetaPtr = [round.store, round.applicationStore];
-
     const initRoles = [
       [await signerOrProvider.getAddress()],
       round.operatorWallets,
     ];
 
-    // Ensure tokenAmount is normalized to token decimals
-    const tokenAmount =
-      round.roundMetadata.quadraticFundingConfig?.matchingFundsAvailable || 0;
-    const token = payoutTokens.filter(
-      (t) => t.address.toLocaleLowerCase() == round.token.toLocaleLowerCase(),
-    )[0];
-    const parsedTokenAmount = utils.parseUnits(
-      tokenAmount.toString(),
-      token.decimal,
-    );
+    const token = ethers.constants.AddressZero;
+    let parsedTokenAmount = ethers.constants.Zero;
+
+    if (isQF) {
+      // Ensure tokenAmount is normalized to token decimals
+      const tokenAmount =
+        round.roundMetadata.quadraticFundingConfig?.matchingFundsAvailable || 0;
+      const pyToken = payoutTokens.filter(
```

Comment:
Was this meant to be `payToken`?

Reply:
This code was like so before, It refers to the tokens that can be selected on the creating round form.